scripts/typescript/afrl-to-new-api/src/scripts/afrl-csv-to-json.ts

```diff
@@ -0,0 +1,631 @@
+                
+                csv_raw_index: index + 1, // one-based index, because the CSV column names use the raw 0.
+
+                reference: raw_object.reference,
+                one_phase_direction: raw_object.one_phase_direction,
+                polymer_CAS: raw_object.polymer_CAS,
+                polymer_SMILES: raw_object.polymer_SMILES,
+                polymer: raw_object.polymer,                
+                solvent_CAS: raw_object.solvent_CAS,
+                solvent_SMILES: raw_object.solvent_SMILES,
+                solvent: raw_object.solvent,
+                ...{
+                    solvent_Mw: to_number_or_undefined(raw_object.solvent_Mw),
+                    polymer_Mw: to_number_or_undefined(raw_object.polymer_Mw),
+                    cloud_point_temp: to_number_or_undefined(raw_object.cloud_point_temp),
+                    mixture_id: to_number_or_undefined(raw_object.mixture_id),
+                    polymer_id: to_number_or_undefined(raw_object.polymer_id),
+                    polymer_PDI: to_number_or_undefined(raw_object.polymer_PDI),
+                    polymer_vol_frac: to_number_or_undefined(raw_object.polymer_vol_frac),
+                    polymer_wt_frac: to_number_or_undefined(raw_object.polymer_wt_frac),
+                    pressure_MPa: to_number_or_undefined(raw_object.pressure_MPa),
+                }
+            } as Partial<AFRLData> as any;
+            return safe_object;
+        });
+        console.log(`Assigning default value for string properties OK`);
+        return afrl_data;
+    }
+
+    /**
+     * Loads structured AFRLData[] into a project
+     * @param data 
+     * @returns 
+     */
+    load_data(data: AFRLData[]): IProject {
+
+        // load data
+        console.log('Loading data ...')
+        const failed_rows: AFRLData[] = [];
+        let one_based_index = 1;
+        const data_length = data.length;
+        for (let row of data) {
+            console.log(`Loading data ${one_based_index}/${data_length} ...`)
+            if (!this.load_row(row)) {
+                failed_rows.push(row);
+            }
+            one_based_index++;
+        }
+
+        // Log failures
+        if (failed_rows.length != 0) {
+            console.log(`Loading failed. Some objects couldn't be loaded (${failed_rows.length} row(s) failed)`)
+            failed_rows.forEach(v => console.error(JSON.stringify(v)))
+            throw new Error(`${failed_rows.length} row(s) were not loaded.`)
+        }
+
+        // hack
+        // In order to avoid to push multiple times the same Node, we have to
+        // set a "uid" (not "uuid"), which is like a local id.
+        const assign_uid = (material: IMaterial) => material.uid = `afrl-csv-to-json:${this.generate_uid()}`;
+        this.solvents.forEach(assign_uid)
+        this.mixtures.forEach(assign_uid)
+        this.polymers.forEach(assign_uid)
+
+        console.log('Loading data OK')
+        return this.project;
```

Comment:
Major issue, no materials are added to the Project. They are only added to their respective Inventory.